### PR TITLE
Fix a bug where the collected event was registered too late

### PR DESCRIPTION
**Pull request by kraih**
Not actually a fix for the issue i'm working on, but a small race condition i found along the way.

Progress: https://progress.opensuse.org/issues/107998

### CODE REVIEW

File: backend/driver.pm [hunk 1]
```diff
@@ -52,8 +53,7 @@ sub start ($self) {
         blocking_stop => 1,
         separate_err => 0,
         subreaper => 1,
-        code => sub {
-            my $process = shift;
+        code => sub ($process) {
```
Comment (Martchus) [inline backend/driver.pm:56]: Judging by RWP code additional arguments might be passed to the callback:
```suggestion
        code => sub ($process, @args) {
```

**Review by Martchus — COMMENTED** [review]
Looks good, except for the one remark mentioned in the inline comment.

**Review by Martchus — APPROVED** [review]

**Review by okurz — CHANGES_REQUESTED** [review]
https://github.com/os-autoinst/os-autoinst/runs/5681532735?check_suite_focus=true#step:3:594 fails in 18-qemu-options.t with an error message I don't really understand.

**Comment by kraih** [comment]
I've restarted the tests a few times, but `18-qemu-options.t` fails consistently in CI. Locally i'm not seeing the problem. Probably a race condition that becomes more common with the patch. I'll investigate and look for a fix.

**Comment by Martchus** [comment]
Strange. Maybe the number of arguments is diverting in certain cases?

**Comment by kraih** [comment]
Maybe i'll try to ssh into the GitHub Actions VM on Monday to track it down further.

**Review by okurz — APPROVED** [review]

**Pull request merged**